### PR TITLE
feat(api): update moduleOffsets when loading/calibrating modules to get accurate labware position.

### PR DESCRIPTION
**Pull request by vegano1**
# Overview

We need to include the calibrated module offsets when calculating the labware position, this way that position is accurate for operations required by the system. This pr will hook in the calibrated data and load it from the disk if it exists, as well as update the existing calibration data when we calibrate a module.

Cloeses: [RCORE-666](https://opentrons.atlassian.net/browse/RCORE-666)

# Test Plan

- [x] Make sure we can still run the `LoadModule` protocol engine command
- [x] Run the `LoadModule` PE command and make sure we are loading module offsets from disk and returning them in `LoadModuleResult`
- [x] Run `CalibrateModule` PE command to get new calibration values, then run `LoadModule` PE command and make sure the `vectorOffset` value in `LoadModuleResult` returns the new calibrated value.

# Changelog

- Load module calibration when creating protocol engine StateStore
- Handle `CalibrateModule` to update the module calibration value
- Add calibrated offset when calculating the module offsets in the modules.states.get_module_offset function

# Review requests

- Is this the right approach?
-  Am I missing something?

# Risk assessment
- Low, flex work

# Todo

- fix unit tests

[RCORE-666]: https://opentrons.atlassian.net/browse/RCORE-666?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/src/opentrons/protocol_engine/state/modules.py [hunk 1]
```diff
@@ -608,6 +631,20 @@ def get_module_offset(
...
+            module_offset = array(
+                (
+                    offset.x,
+                    offset.y,
+                    offset.z,
+                    1
+                )
+            )
+            pre_transform = add(pre_transform, module_offset)
+
```
Comment (sanni-t) [inline api/src/opentrons/protocol_engine/state/modules.py]: I am pretty sure this is not the right way to incorporate the module calibrated offset. The offset that `calibrateModule` calculates would be on top of the value calculated by `get_module_offset`. So we just have to add the calibrated offset to the `LabwareOffsetVector` on line 657.

File: api/src/opentrons/protocol_engine/state/modules.py [hunk 2]
```diff
@@ -165,6 +172,13 @@ def _handle_command(self, command: Command) -> None:
                 slot_name=command.params.location.slotName,
                 requested_model=command.params.model,
                 module_live_data=None,
+                module_offset=command.result.offsetVector,
+            )
+
+        if isinstance(command.result, CalibrateModuleResult):
+            self._update_module_calibration(
+                module_id=command.params.moduleId,
+                module_offset=command.result.moduleOffset
```
Comment (sanni-t) [inline api/src/opentrons/protocol_engine/state/modules.py]: If I understand correctly, module calibration offsets will be saved by the hardware controller onto disk, and then read back from disk using `ardware_control.modules.module_calibration.load_module_calibration_offset`. Then I don't think saving it to the engine like this is necessary at all. In fact, the engine that will perform the calibration and find the module offset will almost certainly not be the engine that will run a protocol and need to use the offset. So caching the value in the engine serves very little purpose.

**Review by sanni-t — CHANGES_REQUESTED** [review]
I think this way of including the offsets in `loadLabwareResult` and caching offsets after a `calibrateModule` command is quite a roundabout way of adding offsets. Also, unless I'm mistaken as to how `module_id` is used, it looks like you'll never be able to load a module offset from disk because the `module_id` generated by the engine is completely random and changes every run.
But before that, I have a basic question- 
Do we intend to calibrate a module before each protocol run, like LPC, or is the plan to calibrate the module once, like an instrument calibration, save that info to disk, and then use those calibrated offsets each time that module is used in a protocol? Also, do we want to save unique offsets for a module+slot combo or do we want to use the same calibrated offset of a module for all slot locations?

File: api/src/opentrons/protocol_engine/state/modules.py (hunk 2 above)
Comment (vegano1) [reply api/src/opentrons/protocol_engine/state/modules.py]: We spoke about this on slack and decided its the correct way of handling the calibration offsets.

**Review by sanni-t — APPROVED** [review]
Loading and applying offsets in engine looks good!

**Pull request merged**